build/config/pom.xml

```diff
@@ -66,6 +66,48 @@
 
     <build>
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-enforcer-plugin</artifactId>
+                <executions>
+                    <!--
+                        We cannot include this execution in the root pom because it depends on a project artifact
+                        that we need to build first.
```

Comment:
Maybe this is an argument in favor of a dedicated "build" parent POM that is not the root POM, and could thus depend on `hibernate-search-build-enforcer`? Assuming `hibernate-search-build-enforcer` does not need all our dependency/plugin configuration...